extensions/gitpod-web/src/extension.ts

```diff
@@ -64,6 +67,22 @@ export function deactivate() {
 	return gitpodContext.dispose();
 }
 
+function logContextInfo(context: GitpodExtensionContext) {
+	const output = context.output;
+
+	const contextUrl = process.env['GITPOD_WORKSPACE_CONTEXT_URL'];
```

Comment:
We should never rely on these env vars. They are Interfaxe between workspace manager and supervisor and can be changed by workspace team. Always use supervisor info endpoint. How about creating something like design checklist for IDE team to review PR against?

Can we write them in frontend? If ext host does not start we loose this info. Workbench.ts should already call supervisor info endpoint.

Reply:
What about the error we are getting reports where connection to server fails? it doesn't seem that's reliable either

> Can we write them in frontend? If ext host does not start we loose this info. Workbench.ts should already call supervisor info endpoint.

I don't have access to server connection here unless I create another connection 🤔

Reply:
I'm not sure what you mean you already have access to all info from supervisor like `context.info.getWorkspaceContextUrl()`

Reply:
Can we write this info both in frontend and in extension host? Instance ID is super crucial to look at exact workspace instance. I don't want to miss it.